test-samples.cpp

```diff
@@ -0,0 +1,70 @@
+#define CATCH_CONFIG_MAIN  // This tells Catch to provide a main() - only do this in one cpp file
+
+#include "test/catch.hpp"
+
+#include "FormatOutput.h"
+#include "DetectChargeRange.h"
+#include "A2DConversion.h"
+
+TEST_CASE("Test charge samples ranges") {
+ 
+   formatOutputValue.formatOutput = formatOutput;
```

Comment:
Good set of test values for the tested APIs 👍

Reply:
When the variety of inputs is more, you may want to separate the testing of the A2D part and the range-counting part ->
Updated review comment